### PR TITLE
Fix to failing Vagrant script

### LINKED ISSUE #64: Vagrant script for ubuntu fails when run in an isolated environment
Standing up an environment using `vagrant` (for ubuntu 14.04) and running the [ubuntu.yml](https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/ansible/playbooks/ubuntu.yml) Ansible script halts with the below message:

```
fatal: [localhost]: FAILED! => {"failed": true, "msg": "An unhandled exception occurred while
running the lookup plugin 'file'. Error was a <class 'ansible.errors.AnsibleError'>, 
original message: could not locate file in lookup: /Vendor_Files/keys/id_rsa.pub"})
```

This occurs when run on a local machine (reproducible on Linux and MacOSX environments).

Re-running with `-v` flags will help, `-vv`, `-vvv`, etc... will give will more verbose info about the issue.

See https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/58#issuecomment-347149275 in https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/58

Add any additional info to https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/ansible/README.md, once resolved or any findings during the course of the investigation.

#helpwanted #bug

### PR DESCRIPTION
**Pull request by neomatrix369**
- Added some more Ansible related documentation. Docs also say how to run a playbook, skip tasks by using the --skip-tags flag and specify the tag names of the tasks to skip
- Added tags to tasks that are necessary for the moment
- Added multiple tags on certain tasks to gain granularity of the steps in the task to point to them

Follow instructions, create and setup vagrant box, Inside the vagrant box, run:
```
ansible-playbook -i /path/to/hosts -s ubuntu.yml --skip-tags "install_zulu,jenkins_authorized_key,nagios_add_key,add_zeus_user_key"
```

Added "install_zulu", as I was still running into the Azul issue with the latest changes, in case one of you already has a fix, I'll be happy to rebase my repo.

Fixes #64

### CODE REVIEW

**Comment by neomatrix369** [comment]
If there is another way to do it, I'll be happy to know.

**Comment by bblondin** [comment]
Each of the changes you have proposed in the ```ubuntu.yml``` playbook would also have to be applied to all other playbooks to allow the same fuction to work everywhere. 

Another why to do this would be to use ```--extra-vars``` as we do in AWX (Ansible-Tower).
Although the name says 'extra' it will also override variables defined inside the playbook.

The command line would looks like this:

```
ansible-playbook -i /path/to/hosts -s ubuntu.yml --extra-vars "Jenkins_Username=MY_USER Jenkins_User_SSHKey=/PATH/id_rsa.pub Nagios_Plugins=Disabled Slack_Notification=Disabled Superuser_Account=Disabled"
```

Editting the ```/etc/ansible/hosts``` file isn't required. This is a global setting.
Another option is to create a folder to work out of it, adding the ```hosts``` file and any other config, along with your playbook. ex:

```
/my_ansible_temp/
-- hosts
-- ansible.cfg 
-- My_Play.yml
```

The ```hosts``` file could look like this

```
My_Host ansible_ssh_host=HOSTNAME_or_IPADDRESS ansible_ssh_pass=MYPASSWORD
```

Warning: ^this stores the password in plain text on the local machine

or you could use ```--ask-pass``` or have perviously setup ssh keys to that machine

**Comment by neomatrix369** [comment]
Thanks @bblondin  I will give it a whirl, and change the PR accordingly.

@karianna or someone else, just for a sanity check, could you pls try to run the script and see if it works for you once I have a new update in.

**Comment by karianna** [comment]
@neomatrix369 I still get:

```
TASK [Install zulu-9-amd64] *************************************************************************************************************************************************************************************************************
failed: [localhost] (item=[u'zulu-9-amd64']) => {"changed": false, "failed": true, "item": ["zulu-9-amd64"], "msg": "No package matching 'zulu-9-amd64' is available"}
```

**Comment by neomatrix369** [comment]
@karianna try running with the below:
```
ansible-playbook -i hosts -s ubuntu.yml --extra-vars "Jenkins_Username=MY_USER Jenkins_User_SSHKey=/PATH/id_rsa.pub Nagios_Plugins=Disabled Slack_Notification=Disabled Superuser_Account=Disabled" --skip-tags="install_zulu"
```

Although we will then need to supply it with a `/PATH/ida_rsa.pub` key or you will get the below: 
```
TASK [Set authorized key for Jenkins user] *******************************************************************************************************
 [WARNING]: Unable to find '/PATH/id_rsa.pub' in expected paths.

fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'file'. Error was a <class 'ansible.errors.AnsibleError'>, original message: could not locate file in lookup: /PATH/id_rsa.pub"}
```

**Comment by neomatrix369** [comment]
@bblondin @karianna 

Just created an ssh public-private key-pair in the vagrant box using:
```
ssh-keygen -t rsa -b 4096 -C "adoptopenjdk@gmail.com"
```
And re-ran the above script like this:

```
ansible-playbook -i hosts -s ubuntu.yml --extra-vars "Jenkins_Username=jenkins Jenkins_User_SSHKey=/home/vagrant/.ssh/id_rsa.pub Nagios_Plugins=Disabled Slack_Notification=Disabled Superuser_Account=Disabled" --skip-tags="install_zulu"
```
and got

```

PLAY RECAP ***************************************************************************************************************************************
localhost                  : ok=38   changed=23   unreachable=0    failed=0 
```

I hope this works for you, if it does, I can remove some of the added tags and amend the docs.

**Comment by neomatrix369** [comment]
The `ubuntu.yml` playbook was built in the below time:
real	11m9.965s
user	6m26.744s
sys	2m15.952s

**Comment by karianna** [comment]
@neomatrix369 I got

```
[DEPRECATION WARNING]: The sudo command line option has been deprecated in favor of the "become" command line arguments. This feature will be removed in version 2.6. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
 [WARNING]: Unable to parse /vagrant/playbooks/hosts as an inventory source

 [WARNING]: No inventory was parsed, only implicit localhost is available

 [WARNING]: Could not match supplied host pattern, ignoring: all

 [WARNING]: provided hosts list is empty, only localhost is available
```

**Comment by neomatrix369** [comment]
@karianna you ran the below:
```
ansible-playbook -i hosts -s ubuntu.yml --extra-vars "Jenkins_Username=jenkins Jenkins_User_SSHKey=/home/vagrant/.ssh/id_rsa.pub Nagios_Plugins=Disabled Slack_Notification=Disabled Superuser_Account=Disabled" --skip-tags="install_zulu"
```
and actually got:
```
[DEPRECATION WARNING]: The sudo command line option has been deprecated in 
favor of the "become" command line arguments. This feature will be removed in 
version 2.6. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
 [WARNING]: Unable to parse /vagrant/playbooks/hosts as an inventory source

 [WARNING]: No inventory was parsed, only implicit localhost is available

 [WARNING]: Could not match supplied host pattern, ignoring: all

 [WARNING]: provided hosts list is empty, only localhost is available


PLAY [all] *********************************************************************
skipping: no hosts matched

PLAY RECAP *********************************************************************
``` 
because you didn't point to the correct `hosts` file, there was no `hosts` in the directory where your ran this command.

I would suggest create a `hosts` file in the `/vagrant/playbooks` folder (inside the VM), it should contain the same contents as mentioned in the README docs.

I should have mentioned that step about the `hosts` file again.

**Comment by karianna** [comment]
@neomatrix369 That works for me now thanks. I think we should update that doc as well to install the hosts file where you suggested.

**Comment by neomatrix369** [comment]
WIll apply these changes to the other playbooks where applicable i.e. 

- aix.yml
- centos_7.yml
- rhel.yml
- sles.yml
- ubuntu.yml
- windows.yml

via another PR.

File: ansible/createSSHKeyPair.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```
Comment (karianna) [inline ansible/createSSHKeyPair.sh]: Can we run this as part of the vagrant?

File: ansible/playbooks/hosts [hunk 2]
```diff
@@ -0,0 +1 @@
+localhost ansible_connection=local
```
Comment (karianna) [inline ansible/playbooks/hosts]: Can we add this via vagrant?

**Comment by neomatrix369** [comment]
@karianna Both those can be done. let me amend the scripts and give it a whirl

**Comment by neomatrix369** [comment]
@karianna Done! Can you pls run them and see if works for you too

**Pull request merged**